commcare_export/checkpoint.py

```diff
@@ -124,25 +135,38 @@ def get_time_of_last_checkpoint(self, log_warnings=True):
         return run.since_param if run else None
 
     def get_last_checkpoint(self):
+        self._validate_tables()
+        table_runs = []
         with session_scope(self.Session) as session:
-            if self.key:
-                run = self._get_last_checkpoint(
-                    session, key=self.key,
-                    project=self.project, commcare=self.commcare
-                )
-            else:
-                run = self._get_last_checkpoint(
-                    session, query_file_md5=self.query_md5,
-                    project=self.project, commcare=self.commcare, key=self.key
-                )
-                if not run:
-                    # Check for run without the args
-                    run = self._get_last_checkpoint(session, query_file_md5=self.query_md5, key=self.key)
-        return run
-
-    def _get_last_checkpoint(self, session, **filters):
-        return session.query(Checkpoint).filter_by(**filters)\
-            .order_by(Checkpoint.time_of_run.desc()).first()
+            for table in self.table_names:
+                if self.key:
+                    table_run = self._get_last_checkpoint(
+                        session, table_name=table,
+                        key=self.key, project=self.project, commcare=self.commcare
+                    )
+                else:
+                    table_run = self._get_last_checkpoint(
+                        session, table_name=table,
+                        query_file_md5=self.query_md5, project=self.project, commcare=self.commcare, key=self.key
+                    )
+                    if not table_run:
+                        # Check for run without the args
+                        table_run = self._get_last_checkpoint(
+                            session, table_name=table,
+                            query_file_md5=self.query_md5, key=self.key
+                        )
+                table_runs.append(table_run)
+
+        sorted_runs = list(sorted(filter(None, table_runs), key=attrgetter('time_of_run')))
+        return sorted_runs[0] if sorted_runs else None
+
+    def _get_last_checkpoint(self, session, *criterion_filters, **kwarg_filters):
```

Comment:
At one point I was using these to do an `in` filter: `Checkpoint.table_name.in_(self.table_names)` but moved away from that approach. I'll remove that for now since it's not used.